src/MoBi.Presentation/Tasks/Interaction/InteractionTasksForSimulation.cs

```diff
@@ -31,7 +32,9 @@ public interface IInteractionTasksForSimulation : IInteractionTasksForChildren<M
       /// <summary>
       ///    Returns the project building block that matches <paramref name="buildingBlock" />
       /// </summary>
-      IBuildingBlock TemplateBuildingBlockFor(IBuildingBlock buildingBlock);
+      TBuildingBlock TemplateBuildingBlockFor<TBuildingBlock>(TBuildingBlock buildingBlock) where TBuildingBlock : class, IBuildingBlock;
+
+      Module TemplateModuleFor(Module module);
```

Comment:
See here, we added some code to the interaction task for simulation (which is a task realted to simulation stuff). This is where I  would consider moving some of the code from above (which also called the InteractionTask for Simulation). Not sure if this makes sense